xaitk_saliency/utils/masking.py

```diff
@@ -83,3 +84,35 @@ def generate_masked_images(
         masked_img = np.multiply(mask, img, casting='unsafe')
         masked_imgs.append(PIL.Image.fromarray(np.uint8(masked_img)))
     return masked_imgs
+
+
+def weight_regions_by_scalar(
+    scalar_vec: np.ndarray,
+    masks: np.ndarray
+) -> np.ndarray:
+    """
+    Weight some binary masks region with its respective vector in scalar_vec.
+
+    We expect the "masks" matrices and the image to be the same height and
+    width, and be valued in the [0, 1] floating-point range. The length
+    and order of scalar_vec and masks are assumed to be the same.
+    In the mask matrix, higher values correspond to regions of the image that
+    preserved. E.g. a 0 in the mask will translate to blacking out the
+    corresponding location in the source image.
+
+    :param scalar_vec: Weights for image regions for nClasses and shape
+                       [nMasks, nClasses]
+    :param masks: Mask array in the [nMasks, Height, Weight] shape format.
+
+    :return: A numpy array representing the weighted heatmap.
+    """
+    # Weighting each perturbed region with its respective score in vector.
+    heatmap = (np.expand_dims(np.transpose(1 - masks), axis=3) * scalar_vec)
+
+    # Aggregate scores across all perturbed regions.
+    sal_across_masks = np.transpose(heatmap.sum(axis=2))
+
+    # Compute final saliency map by normalizing with sampling factor.
+    final_heatmap = sal_across_masks/masks.sum(axis=0)
```

Comment:
@nitesh201 Please take a look at the newer version accounting for any unequal sampling by normalizing the final saliency map with the aggregate of all perturbed masks. Feel free to give this a try with RISE, and please let me know if you come across issues.

Reply:
Per our offline discussion, line 115 should be

```
final_heatmap = sal_across_masks/(1 - masks).sum(axis=0)
```

Otherwise, everything else LGTM!